src/runtime/virtcontainers/pkg/agent/protocols/client/client.go

```diff
@@ -62,12 +62,17 @@ type dialer func(string, time.Duration) (net.Conn, error)
 //     model, and mediates communication between AF_UNIX sockets (on the host end)
 //     and AF_VSOCK sockets (on the guest end).
 //   - mock://<path>. just for test use.
-func NewAgentClient(ctx context.Context, sock string) (*AgentClient, error) {
+func NewAgentClient(ctx context.Context, sock string, timeout uint32) (*AgentClient, error) {
 	grpcAddr, parsedAddr, err := parse(sock)
 	if err != nil {
 		return nil, err
 	}
 
+	if timeout > 0 {
+		agentClientLog.WithField("timeout", timeout).Debug("Default dialing timeout has changed")
+		defaultDialTimeout = time.Duration(timeout) * time.Second
+	}
+
```

Comment:
Nit, Suggest this:

```golang
if timeout <= 0 {
    timeout = defaultDialTimeout
}

conn, err = d(grpcAddr, timeout)
```

Reply:
Hi @liubin the thing is types are different, just pushed another option